src/PrestaShopBundle/Api/QueryParamsCollection.php

```diff
@@ -245,16 +255,18 @@ protected function removeDirection($subject)
      */
     public function getSqlOrder()
     {
-        $descendingOrder = false !== strpos($this->queryParams['order'], 'desc');
-        $filterColumn = $this->removeDirection($this->queryParams['order']);
+        foreach ($this->queryParams['order'] as &$order) {
```

Comment:
This loop produces side effects (`$this->queryParams` is changed).

It's not a good idea to have side effects in getters. What happens if you call it twice?